boptimol/GP_MODEL.py

```diff
@@ -0,0 +1,239 @@
+from typing import List, Optional
+import torch
+import gpytorch
+import numpy as np
+
+from botorch.utils import standardize
+from botorch.optim import optimize_acqf
+from botorch.acquisition.analytic import *
+from botorch.models import SingleTaskGP
+from botorch.fit import fit_gpytorch_mll
+from botorch.acquisition import qKnowledgeGradient
+from gpytorch.mlls import ExactMarginalLogLikelihood
+from gpytorch import kernels as gpykernels
+from gpytorch import means as gpymeans
+
+from boptimol.molecule import Molecule
+from boptimol.mean_potentials import MorseMean, QuadraticMean
+
+class Bond_Lengths(gpytorch.models.ExactGP):
+    def __init__(self, train_x, train_y, likelihood):
+        super(Bond_Lengths, self).__init__(train_x, train_y, likelihood)
+        self.mean_module = gpytorch.means.ConstantMeanGrad()
+        self.base_kernel = gpytorch.kernels.RBFKernelGrad()
+        self.covar_module = gpytorch.kernels.ScaleKernel(self.base_kernel)
+
+    def forward(self, x):
+        mean_x = self.mean_module(x)
+        covar_x = self.covar_module(x)
+        return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)
+
+
+def select_next_point_bond_lengths_derivative(bounds: torch.Tensor,
+        observed_X: List[List[float]], 
+        observed_y: List[float],
+        device: str) -> np.ndarray:
+    
+    """Generate the next sample to evaluate
+
+    Uses BOTorch to pick the next sample using Expected Improvement
+
+    Args:
+        observed_X: Observed coordinates
+        observed_y: Observed energies
+    Returns:
+        Next coordinates to try
+    """
+    
+    likelihood = gpytorch.likelihoods.MultitaskGaussianLikelihood(num_tasks=1).to(device=device)
+    gp = Bond_Lengths(observed_X, observed_y, likelihood)
+    mll = gpytorch.mlls.ExactMarginalLogLikelihood(gp.likelihood, gp)
+    fit_gpytorch_mll(mll)
+    
+    qKG = qKnowledgeGradient(gp, num_fantasies=4)
```

Comment:
This seems like you're experimenting with KnowledgeGradient, which is great. But then why have the logEI below that?

I think we need to have a way to pass or set the acquisition function for all these classes - so that we can easily benchmark the different acquisition methods.